File: app/src/main/java/io/intrepid/contest/base/BaseContract.java
package io.intrepid.contest.base;

import android.support.annotation.StringRes;

public class BaseContract {

    public interface View {
        void showMessage(String message);

        void showMessage(@StringRes int messageResource);
    }

    public interface Presenter<T extends View> {

        void bindView(T view);

        void unbindView();

        void onViewCreated();

        void onViewDestroyed();
    }
}

File: app/src/main/java/io/intrepid/contest/base/BaseContract.kt
package io.intrepid.contest.base

import android.support.annotation.StringRes

class BaseContract {

    interface View {
        fun showMessage(message: String)

        fun showMessage(@StringRes messageResource: Int)
    }

    interface Presenter<in T : View> {

        fun bindView(view: T)

        fun unbindView()

        fun onViewCreated()

        fun onViewDestroyed()
    }
}
